### ISSUE TITLE
ReLu Operation and Binarization function

### ISSUE THREAD

**Issue by yaduydk97** [jiecaoyu/XNOR-Net-PyTorch#89]
1.Are you using ReLu Activation function in every layer of AlexNet?? If so,what's its significance and where are we using it,Is it used just before maxpooling step ?
2.Are you not binarizing the input for 1st layer??,I think in all other layers,you are binarizing the input.
3.Please,be specific why you are not binarizing 1st layer,if so??
4.Is batch nornalisation necessaryHow it is done and,where in the code are you doing that?

**Comment by jiecaoyu** [comment]
@yaduydk97 
1. I am applying ReLU after the ```Convolution```. It is mentioned that ```We can insert a non-binary activation(e.g.,ReLU) after binary convolution.``` in the original paper (page.9).

2. No. I didn't binarize the input for the first and last layer.

3. It hurts accuracy a lot.

4. Yes. It is the same as the conventional batch norm. BN is applied [here](https://github.com/jiecaoyu/XNOR-Net-PyTorch/blob/ebccb503b2f9ebc96187a75b6daa63d95b61d110/ImageNet/networks/model_list/alexnet.py#L50).

**Comment by yaduydk97** [comment]
Thank you @jiecaoyu
